File: university-service/src/main/java/ua/com/foxminded/courseproject/exceptions/StudentNotFoundException.java
package ua.com.foxminded.courseproject.exceptions;

import java.util.NoSuchElementException;
import java.util.UUID;

public class StudentNotFoundException extends NoSuchElementException {

    static private final String message = "Can't find student with ID=%s";

    public StudentNotFoundException() {
        super();
    }

    public StudentNotFoundException(String s, Throwable cause) {
        super(s, cause);
    }

    public StudentNotFoundException(Throwable cause) {
        super(cause);
    }

    public StudentNotFoundException(String s) {
        super(s);
    }

    public StudentNotFoundException(UUID s) {
        super(message.formatted(s.toString()));
    }
}

File: university-service/src/main/kotlin/ua/com/foxminded/courseproject/exceptions/StudentNotFoundException.kt
package ua.com.foxminded.courseproject.exceptions

import java.util.UUID

class StudentNotFoundException : NoSuchElementException {
    constructor() : super()
    constructor(s: String, cause: Throwable) : super(s, cause)
    constructor(cause: Throwable) : super(cause)
    constructor(s: String) : super(s)
    constructor(s: UUID) : super(MESSAGE.format(s.toString()))

    companion object {
        private const val MESSAGE = "Can't find student with ID=%s"
    }
}
